File: Server_Nodejs/models/danhmuc.js
import { DataTypes } from 'sequelize';
import mydb from '../config/connect_mysql.js';

const DanhMuc = mydb.define('danhmuc', {
    id_danhmuc: {
        type: DataTypes.INTEGER,
        primaryKey: true,
        autoIncrement: true
    },
    ten_danhmuc: {
        type: DataTypes.STRING,
        allowNull: true
    },
    url_image: {
        type: DataTypes.STRING,
        allowNull: true
    },
    status: {
        type: DataTypes.BOOLEAN,
        allowNull: false
    }
}, {
    tableName: 'danhmuc',
    timestamps: false
});

export default DanhMuc;

File: second_hand_store/lib/models/danhmuc.dart
class DanhMuc {
  int? idDanhmuc;
  String? tenDanhmuc;
  bool? status;

  DanhMuc({this.idDanhmuc, this.tenDanhmuc, this.status});

  DanhMuc.fromJson(Map<String, dynamic> json) {
    idDanhmuc = json['id_danhmuc'];
    tenDanhmuc = json['ten_danhmuc'];
    status = json['status'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id_danhmuc'] = idDanhmuc;
    data['ten_danhmuc'] = tenDanhmuc;
    data['status'] = status;
    return data;
  }
}
